gc3libs/workflow.py

```diff
@@ -401,7 +401,10 @@ def redo(self, from_stage=0, *args, **kwargs):
             task.redo(*args, **kwargs)
         # All other tasks should be put in NEW again
         for i in range(from_stage+1, len(self.tasks)):
-            self.tasks[i].execution.state = Run.State.NEW
+            try:
+                self.tasks[i].redo(*args, **kwargs)
+            except IndexError:
```

Comment:
Because I got:

```
  File "/Users/mdh/gc3pie/gc3libs/core.py", line 1694, in redo
    task.redo(*args, **kwargs)
  File "/Users/mdh/gc3pie/gc3libs/workflow.py", line 403, in redo
    task.redo(*args, **kwargs)
  File "/Users/mdh/gc3pie/gc3libs/workflow.py", line 403, in redo
    task.redo(*args, **kwargs)
  File "/Users/mdh/gc3pie/gc3libs/workflow.py", line 409, in redo
    self.tasks[i].redo(*args, **kwargs)
  File "/Users/mdh/gc3pie/gc3libs/workflow.py", line 401, in redo
    task = self.stage()
  File "/Users/mdh/gc3pie/gc3libs/workflow.py", line 443, in stage
    return self.tasks[self._current_task]
IndexError: list index out of range
```